handlers/handlers_test.go

```diff
@@ -0,0 +1,243 @@
+					return codelist.CodesResults{
+						Items: []codelist.Item{
+							codelist.Item{
+								ID:    "E06000028",
+								Label: "Bournemouth",
+								Links: codelist.CodeLinks{},
+							},
+							codelist.Item{
+								ID:    "S12000033",
+								Label: "Aberdeen City",
+								Links: codelist.CodeLinks{},
+							},
+							codelist.Item{
+								ID:    "S12000034",
+								Label: "Aberdeenshire",
+								Links: codelist.CodeLinks{},
+							},
+						},
+						Count:      2,
+						TotalCount: 2,
+						Offset:     0,
+						Limit:      2,
+					}, nil
+				},
+			}
+
+			router.Path("/geography/{codeListID}").HandlerFunc(ListPageRender(mockRenderClient, mockCodeListClient))
+			router.ServeHTTP(w, req)
+			renderCall := mockRenderClient.DoCalls()[0]
+
+			var payload list.Page
+			err := json.Unmarshal(renderCall.In2, &payload)
+			if err != nil {
+				t.Errorf("Failed to unmarshal payload sent to renderer to relevant frontend model: %s", err)
+				return
+			}
+
+			So(payload.Metadata.Title, ShouldEqual, "Local authority districts")
+			So(payload.Breadcrumb, ShouldResemble, []model.TaxonomyNode{
+				model.TaxonomyNode{
+					Title: "Home",
+					URI:   "https://www.ons.gov.uk",
+				},
+				model.TaxonomyNode{
+					Title: "Geography",
+					URI:   "/geography",
+				},
+				model.TaxonomyNode{
+					Title: "Local authority districts",
+					URI:   "/geography/local-authority",
+				},
+			})
+			So(len(payload.Data.Items), ShouldEqual, 3)
+			So(payload.Data.Items, ShouldResemble, []list.Item{
+				list.Item{
+					ID:    "S12000033",
+					Label: "Aberdeen City",
+					URI:   "/geography/local-authority/S12000033",
+				},
+				list.Item{
+					ID:    "S12000034",
+					Label: "Aberdeenshire",
+					URI:   "/geography/local-authority/S12000034",
+				},
+				list.Item{
+					ID:    "E06000028",
+					Label: "Bournemouth",
+					URI:   "/geography/local-authority/E06000028",
+				},
+			})
+		})
+
+		Convey("return an error status if request to GET code-list's editions fails", func() {
+			mockRenderClient := &RenderClientMock{
+				DoFunc: func(path string, bytes []byte) ([]byte, error) {
+					return bytes, nil
+				},
+			}
+			mockCodeListClient := &CodeListClientMock{
+				GetCodeListEditionsFunc: func(codeListID string) (codelist.EditionsListResults, error) {
+					return codelist.EditionsListResults{}, errors.New("Code-list %s not found")
+				},
+				GetCodesFunc: func(codeListID string, edition string) (codelist.CodesResults, error) {
+					return codelist.CodesResults{}, errors.New("Code-list %s not found")
+				},
+			}
+
+			router.Path("/geography/{codeListID}").HandlerFunc(ListPageRender(mockRenderClient, mockCodeListClient))
+			router.ServeHTTP(w, req)
+			So(w.Code, ShouldEqual, 500)
+			So(len(mockRenderClient.DoCalls()), ShouldEqual, 0)
+		})
+
+		Convey("return a 500 status if request to GET code-list's codes fails", func() {
```

Comment:
this description is inconsistent with the descriptions of the preceding and following test - either they should all mention the status or none of them should?

Reply:
I've updated all the tests to state what error code that should be returned.